daemon/cmd/agenthealth.go

```diff
@@ -42,6 +42,7 @@ func (d *Daemon) startAgentHealthHTTPService() {
 		statusCode := http.StatusOK
 		sr := d.getStatus(true)
 		if isUnhealthy(&sr) {
+			log.WithField("state", sr.Cilium.State).Warnf("/healthz returning unhealthy: %s", sr.Cilium.Msg)
```

Comment:
Do you think it might be useful to rate limit logs here? I'm concerned that it would spam the logs if the Agent is having flapping issues.

Reply:
@christarazi this should log at the rate that the `/healthz` endpoint is hit by an HTTP client. In the default case I believe that's every 10 seconds by the kubelet liveness check. Probably useful to have that log line at that sort of a rate to make sure that it's obvious there is a health check issue when looking at logs. Happy to add the rate limiting however if you would prefer.

Reply:
Ah, in that case, let's avoid complicating it and see how it goes. We can always fix it up later if it's a problem.